tpl/template_funcs_test.go

```diff
@@ -324,6 +324,30 @@ func TestAfter(t *testing.T) {
 	}
 }
 
+func TestDictionary(t *testing.T) {
+	for i, this := range []struct {
+		v1     []interface{}
+		expecterr bool
+		expectedValue map[string] interface{}
+	}{
+		{[]interface{}{"a", "b"}, false, map[string]interface{}{"a":"b"}},
+		{[]interface{}{5, "b"}, true,nil},
+		{[]interface{}{"a", 12,"b",[]int{1, 2, 4}}, false,map[string]interface{}{"a":12,"b":[]int{1, 2, 4}}},
+		{[]interface{}{"a", "b", "c"}, true,nil},
+	} {
+		r,e := Dictionary(this.v1...)
+
+		if (this.expecterr && e==nil) || (!this.expecterr && e!=nil)  {
+			t.Errorf("[%d] got an unexpected error", i, e, this.expecterr)
+		} else if !this.expecterr {
+			if !reflect.DeepEqual(r, this.expectedValue) {
```

Comment:
The order of the keys is not guaranteed. This has been the situation since Go 1.1. It may work 999 out of 1000 times, maybe all the time on Windows, but some times on Linux ... my point is: Having shaky tests is a bad solution. Esp. when the workaround is known: Sort the map before doing the compare. 

I'm not debating this. Fix it if you want.